refactor: Remove wal-g v3 and simplify build

hunleyd opened the pull request:
- Removes the `wal-g-3` Nix package definition and all associated build/installation steps from Dockerfiles and Ansible playbooks.
- Standardizes on `wal-g-2` as the sole WAL-G version used, simplifying the build process and reducing complexity.
- Cleans up related checks and package definitions in Nix.

hunleyd committed:
refactor: Remove wal-g v3 and simplify build

- Removes the `wal-g-3` Nix package definition and all associated build/installation steps from Dockerfiles and Ansible playbooks.
- Standardizes on `wal-g-2` as the sole WAL-G version used, simplifying the build process and reducing complexity.
- Cleans up related checks and package definitions in Nix.

samrose commented:
if nothing in infra is using this we should be good to go @hunleyd

hunleyd committed:
Merge branch 'develop' of github.com:supabase/postgres into PSQL-822

* 'develop' of github.com:supabase/postgres:
  chore: upgrade supabase-admin-agent (#1935)
  fix: garbage collect after every profile intall + remove osquery (#1938)
  fix(ci): limit max-jobs of nix to 8 to prevent OOM while running nix flake check (#1933)
  Revert "fix: disable temporary Nix checks on aarch64-linux" (#1934)

Crispy1975 commented:
Nothing should be using `wal-g v3` as we never got to the point of migrating, and now are looking to `pgBackRest` in the future.

hunleyd committed:
chore: Add wal-g nix garbage collection

- Executes `nix-collect-garbage -d` after installing wal-g via nix profile.
- This ensures cleanup of the Nix store, freeing up disk space.

hunleyd committed:
Merge remote-tracking branch 'origin/develop' into PSQL-822

* origin/develop:
  refactor: Refactor PostgreSQL cron config and deprecations (#1951)
  chore: bump Wrappers version to 0.5.7 (#1945)
  chore: bump versions (#1949)
  Revert "docs: create an automated changlog file that shows up in release file…" (#1948)
  chore: remove precreated Storage objects (#1911)
  refactor: Move PostgreSQL auto_explain config to conf.d (#1943)
  docs: create an automated changlog file that shows up in release files (#1793)
  fix: removing dead nix code (#1946)
  docs: expansion to help understand anatomy and code api in user here (#1944)
  fix: try to use mirror fallbacks for docker images (#1942)
  feat: need this workflow present to test (#1941)
  feat: Add PostgreSQL conf.d directory (#1929)
  [jj-spr] initial version (#1866)

hunleyd committed:
chore: Update apt cache for Pgbouncer dependencies

Removes `cache_valid_time: 3600` from the apt task for Pgbouncer dependencies.

This ensures that `apt update` is executed every time the playbook runs, guaranteeing the system fetches the latest package lists before installing Pgbouncer and its dependencies. This prevents potential installation failures caused by an outdated package cache.

hunleyd committed:
chore: Update PostgreSQL release versions

Removes the `-INDATA257` suffix from the PostgreSQL release version strings in `ansible/vars.yml`. This aligns the versions closer to standard upstream releases.
